src/app/inventory/store/masterwork.ts

```diff
@@ -92,3 +93,34 @@ function buildMasterworkInfo(
     stats,
   };
 }
+
+/**
+ * Determine if a masterwork with this primary stat would be a valid
+ * masterwork for this item.
+ */
+export function isValidMasterworkStat(
+  defs: D2ManifestDefinitions,
+  itemDef: DestinyInventoryItemDefinition,
+  statHash: number
+) {
+  // Bows have a charge time stat that nobody asked for
+  if (
+    statHash === StatHashes.ChargeTime &&
+    itemDef.itemCategoryHashes?.includes(ItemCategoryHashes.Bows)
+  ) {
+    return false;
+  }
+
+  // Only swords have an impact masterwork
+  if (
+    statHash === StatHashes.Impact &&
+    !itemDef.itemCategoryHashes?.includes(ItemCategoryHashes.Sword)
+  ) {
```

Comment:
curious what this is preventing. do the scaledStats lines not exclude impact?

Reply:
They often don't. From a quick glance at the defs it seems guns with different archetypes and a different damage per bullet between those archetypes include an impact stat in their statgroup.

(Entirely unrelated, has anyone seen a stability masterwork roll on a rocket launcher yet?)

Reply:
![ELzNlkLU8AAD1oV](https://user-images.githubusercontent.com/68782081/151733353-04ffff83-ccd5-4226-8725-6187d8c08d17.jpg)

it's possible... but not sure if it's common. this is a dawn season pass curated roll.